internal/services/solution_checker_rules/client.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package solution_checker_rules
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/microsoft/terraform-provider-power-platform/internal/api"
+)
+
+type client interface {
+	GetSolutionCheckerRules(ctx context.Context, environmentId string) ([]ruleDto, error)
+}
+
+type clientImpl struct {
+	Api *api.Client
+}
+
+func newSolutionCheckerRulesClient(apiClient *api.Client) client {
+	return &clientImpl{
+		Api: apiClient,
+	}
+}
+
+// Data transfer objects.
+type ruleDto struct {
+	Code            string `json:"code"`
+	Description     string `json:"description"`
+	Summary         string `json:"summary"`
+	HowToFix        string `json:"howToFix"`
+	GuidanceUrl     string `json:"guidanceUrl"`
+	ComponentType   int    `json:"componentType"`
+	PrimaryCategory int    `json:"primaryCategory"`
+	Include         bool   `json:"include"`
+	Severity        int    `json:"severity"`
+}
+
+func (c *clientImpl) GetSolutionCheckerRules(ctx context.Context, environmentId string) ([]ruleDto, error) {
+	// Get the environment to find the PowerAppsAdvisor endpoint
+	apiUrl := &url.URL{
```

Comment:
we already have a method that queries environments in environments client. That clients should be referenced and reused to do so, including responseDto's